sycl/doc/extensions/experimental/sycl_ext_intel_fpga_task_sequence.asciidoc

```diff
@@ -0,0 +1,498 @@
+    task_sequence<vectorSum> secondHalf;
+    constexpr int halfCount = kCount/2;
+    firstHalf.async(in.data(), 0, halfCount);
+    secondHalf.async(in.data(), halfCount+1, kCount-1);
+    *out = firstHalf.get() + secondHalf.get();
+  }).wait();
+  ...
+}
+```
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_INTEL_FPGA_TASK_SEQUENCE` to one of the values defined in the
+table below.  Applications can test for the existence of this macro to determine
+if the implementation supports this feature, or applications can test the
+macro's value to determine which of the extension's features the implementation
+supports.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|The APIs of this experimental extension are not versioned, so the
+ feature-test macro always has this value.
+|===
+
+== Definitions
+
+== `task_sequence` class
+
+[source,c++,linenums]
+----
+namespace sycl::ext::intel::experimental {
+
+template <auto &f, class propertiesT>
+class task_sequence;
+
+} // namespace sycl::ext::intel::experimental
+
+----
+
+`task_sequence` is a class template, parameterized by an `auto` reference to a
++Callable+ `f` and by a list of properties `propertiesT`. The +Callable+ `f`
+defines the asynchronous task to be associated with the `task_sequence`, and
+requiring an auto reference ensures that each `f` is statically resolvable at
+compile time. Static resolvability by the compiler is desirable when compiling
+for spatial architectures as it can enable the generation of more efficient
+hardware. Templating a task_sequence on a non-Callable is malformed.
+
+The `task_sequence` interface consists of two member functions, `async` and
+`get`. The `async` member function, when called with a suitable list of
+arguments matching `f's` list of argument types `ArgTypes`, asynchronously
+invokes the `task_sequence` object's associated +Callable+ `f` with those
+arguments. By calling `async` on a `task_sequence` object more than once, the
+user implies that those invocations of `f` can be run in parallel. The
+implementation is, however, not obligated to run the invocations in parallel
+except in so far as is necessary to meet the forward progress guarantees
+outlined in the section on Progress Guarantees.
+
+The `async` function call is non-blocking in that it may return before the
+asynchronous `f` invocation completes executing, and potentially before `f` even
+begins executing (return from the `async` provides no implicit information on
+the execution status of `f`).
+
+The `get` member function, when called, retrieves a return value of type `R`,
+which is the same return type as the `task_sequence` object's associated
++Callable+ `f`. An asynchronous invocation of `f`, upon the end of its execution
+of `f`, will net a return value. At that point, that particular asynchronous
```

Comment:
I'm not familiar with this term: "net a return value"; is this a typo?

Reply:
Reworded